### PR TITLE
LPS-143148 Send external model instead of internal model when sending webhooks notifications

### PR DESCRIPTION
**Pull request by LuismiBarcos**
Related ticket -> [LPS-143148](https://issues.liferay.com/browse/LPS-143148)

# Important note
This is the first PR of 2. In this one, we are changing the behavior of the webhooks as we are modifying the information sent.
The next PR will be a refactor to extract business logic from the `object-api` module and set it in the `object-service` module as we did in [this PR](https://github.com/liferay-frontend/liferay-portal/pull/2082)

# Send external model instead of internal model
The ideal case is to send the model that is specified in the correspondent OpenAPI and that is sent by the Rest APIs so we are consistent with the information Liferay is sharing with external systems.

These are the most important changes:
* Send the external model in addition to the internal model using the DTO converter of the entity. *There are some cases where the internal model will be sent*: 
  * When the DTOConverter for the entity does not exist (There isn't an API with a correspondent external model)
  * When the DTOConverter cannot convert the internal model to the external model for some reason. Normally, will be an error or unsupported case in the DTOConverter that should be fixed, so it is important to log the issue
* Addition of the property  `webhook.expose.internal.model` to specify if we want to send the internal and the external model. By default is `true`
* Webhook sends internal and external models if the property `webhook.expose.internal.model` is set to true.
The new properties have the same prefix as the old ones plus the word `DTO` followed by the name of the entity in the external model. For example, modelUser is modelDTOUserAccount. I maintain the old properties name with the internal model to avoid breaking compatibility.

### How to test this functionality
* Deploy the `object-api`, `object-service` and `headless-admin-user-impl` modules.
* Go to Objects and create webhook actions for an object
* Create/update/delete information of the object selected
* Check the information sent by Liferay to the webhook URL

### CODE REVIEW

**Comment by LuismiBarcos** [comment]
cc/ @gabrielwas

**Comment by LuismiBarcos** [comment]
I'm sending this in two separate PRs as Brian requested because it is easier for him to work with this and to avoid problems if the refactor brokes the webhooks functionality. After Brian merges this, I will send the refactor PR

**Comment by LuismiBarcos** [comment]
ci:test:sf

**Comment by LuismiBarcos** [comment]
ci:test:relevant

File: modules/apps/object/source-formatter-suppressions.xml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0"?>
 
 <suppressions>
-	<checkstyle>
-		<suppress checks="GenericTypeCheck" files="object-api/src/main/java/com/liferay/object/system/model/listener/SystemObjectDefinitionMetadataModelListener\.java" />
-	</checkstyle>
+	<source-check>
```
Comment (javierdearcos) [inline modules/apps/object/source-formatter-suppressions.xml:4]: Why this change?

**Review by javierdearcos — APPROVED** [review]

**Comment by javierdearcos** [comment]
ci:forward

File: modules/apps/object/source-formatter-suppressions.xml (hunk 1 above)
Comment (LuismiBarcos) [reply modules/apps/object/source-formatter-suppressions.xml:4]: Yep, I prefer to avoid suppressions in SF as long as we can code it according to SF. So, in the future, if someone adds more logic (hope no, it is an `api` module 😁 ) the SF will complain in case they add code with generics that should be checked.
In this case, I made this change
```
@Override
	public void onAfterUpdate(BaseModel originalBaseModel, BaseModel baseModel)
```
To
```
@Override
	public void onAfterUpdate(T originalBaseModel, T baseModel)
```
Note the change of `BaseModel` to T. 
This change applies to more than one method.
Also, I add the generic definition at the top of the class from:
```
public class SystemObjectDefinitionMetadataModelListener
	extends BaseModelListener {
```
to
```
public class SystemObjectDefinitionMetadataModelListener<T extends BaseModel<T>>
	extends BaseModelListener<T> {
```

**Comment by javierdearcos** [comment]
Seems we need to rebase to execute baseline in objects-api

**Comment by LuismiBarcos** [comment]
Rebased and updated semver

**Comment by LuismiBarcos** [comment]
ci:forward

**Comment by LuismiBarcos** [comment]
ci:forward

**Comment by LuismiBarcos** [comment]
ci:forward
